src/device/wg_controller.rs

```diff
@@ -151,8 +151,11 @@ async fn consolidate_wg_peers<
     post_quantum_vpn: &impl telio_pq::PostQuantum,
     features: &Features,
 ) -> Result {
-    let proxy_endpoints = if let Some(p) = proxy {
-        p.get_endpoint_map().await?
+    let (proxy_endpoints, replaced_proxy_endpoints) = if let Some(p) = proxy {
+        (
+            p.get_endpoint_map().await?,
+            p.take_replaced_sockets().await?,
```

Comment:
Does the following code ensure, that after this consolidation we will have updated peer to have the new endpoint?

Reply:
What happens if we recreate proxy endpoint multiple times before consolidation?